diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d2667c..27dcab8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,8 +8,8 @@ project(Bifrost)
 # Actual maximum kmer size is 1 less.
 set(MAX_KMER_SIZE "32")
 
-set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} "-std=c11 -march=native -DXXH_NAMESPACE=BIFROST_HASH_")
-set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-std=c++11 -march=native -DXXH_NAMESPACE=BIFROST_HASH_")
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} "-std=c11 -DXXH_NAMESPACE=BIFROST_HASH_")
+set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-std=c++11 -DXXH_NAMESPACE=BIFROST_HASH_")
 
 #set_property(SOURCE xxhash.c APPEND_STRING PROPERTY COMPILE_FLAGS " -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes -Wundef")
 set_property(SOURCE BlockedBloomFilter.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " -funroll-loops")
