diff --git a/setup.py b/setup.py
index d3c7c9f..4af20ac 100755
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,6 @@ import os, platform, sys
 #from distutils.core import setup
 from distribute_setup import use_setuptools 
 import shutil
-use_setuptools(version="0.6.24")
 from setuptools import setup, find_packages    
 from distutils.core import setup, Command
 from distutils.command.install import install
